gcloud-java-compute/src/main/java/com/google/gcloud/compute/DeprecationStatus.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.compute;
+
+import com.google.common.base.Function;
+import com.google.common.base.MoreObjects;
+
+import java.io.Serializable;
+import java.util.Objects;
+
+/**
+ * The deprecation status associated to a Google Compute Engine resource.
+ */
+public final class DeprecationStatus<T extends ResourceId> implements Serializable {
+
+  private static final long serialVersionUID = -2695077634793679794L;
+
+  private final String deleted;
```

Comment:
I think it is much simpler for most people to convert from millisecond epoch rather than from a string (of which you need to know the format, and much easier to get wrong). However, consider it as a nit, as I agree that neither is great and will be changed when we move to Java 8.
BTW, I would argue that consistency with the rest of *our* API is more important in that case.

Sorry for the intermediate review. Will let you when I am done (hopefully by EOD PST).